prog/kubernetes/kubernetes_cluster_nexus.rb

```diff
@@ -16,26 +16,25 @@ def self.assemble(name:, project_id:, location_id:, version: "v1.32", private_su
       Validation.validate_kubernetes_name(name)
       Validation.validate_kubernetes_cp_node_count(cp_node_count)
 
+      ubid = KubernetesCluster.generate_ubid
       subnet = if private_subnet_id
-        project.private_subnets_dataset.first(id: private_subnet_id) || fail("Given subnet is not available in the given project")
+        PrivateSubnet[id: private_subnet_id, project_id: Config.kubernetes_service_project_id] || fail("Given subnet is not available in the k8s project")
       else
-        subnet_name = name + "-k8s-subnet"
-        ps = project.private_subnets_dataset.first(:location_id => location_id, Sequel[:private_subnet][:name] => subnet_name)
-        ps || Prog::Vnet::SubnetNexus.assemble(
-          project_id,
+        subnet_name = ubid.to_s + "-subnet"
+        Prog::Vnet::SubnetNexus.assemble(
+          Config.kubernetes_service_project_id,
           name: subnet_name,
```

Comment:
nit: It has single usage. Do we really need a variable?
```suggestion
          name: "#{ubid}-subnet",
```